R/calculate_study_day.R

```diff
@@ -0,0 +1,142 @@
+#' `calculate_study_day` performs study day calculation
+#' @description
+#' This function takes the an input data.frame and a reference data.frame (which
+#' is DM domain in most cases), and calculate the study day from reference date
+#' and target date. In case of unexpected conditions like reference date is not
+#' unique for each patient, or reference and input dates are not actual dates,
+#' NA will be returned for those records.
+#'
+#' @md
+#' @param sdtm_in input data.frame that contains the target date.
+#' @param dm_domain reference date.frame that contains the reference date.
+#' @param refdt reference date from `dm_domain` that will be used as reference to
+#' calculate the study day.
+#' @param tgdt target date from `sdtm_in` that will be used to calcualte the study
+#' day.
+#' @param study_day_var the new study day variable name in the output. For
+#' example, AESTDY for AE domain for CMSTDY for CM domain.
+#' @param merge_key character to represents the merging key between `sdtm_in` and
+#' `dm_domain`.
+#'
+#' @return a data.frame that takes all columns from `sdtm_in` and a new variable
+#' to represent the calculated study day.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
```

Comment:
Could you confirm why this example can't be run?

Reply:
Oh just my personal preference to not run the examples for R CMD checks. I can remove it if you think we should run these

Reply:
done at https://github.com/pharmaverse/sdtm.oak/pull/20/commits/8a977b2a9baaedf71d4d16f898130c7993263f9c